jsonnet/kube-prometheus/kube-prometheus-thanos-sidecar.libsonnet

```diff
@@ -26,6 +26,15 @@ local servicePort = k.core.v1.service.mixin.spec.portsType;
         ],
       },
     },
+    // Create a new service that exposes both sidecar's HTTP metrics port and gRPC StoreAPI
+    serviceThanosSidecar:
```

Comment:
both of these should just merge into and extend the existing `Service` and `ServiceMonitor` objects

Reply:
I started doing that initially, but I wasn't able to set a different job label on the ServiceMonitor without pointing it to a dedicated service as in this case. Whilst other labels could still be used to separate out the sidecar metrics from the Prometheus metrics, I wanted to try and maintain drop in compatibility with the dashboards and alerts that the Thanos Mixin provides - which rely on the job label to uniquely identify the sidecar's metrics.

If there is a way to apply a `job=thanos-sidecar` label to metrics from an extended and unified service object though, rather than splitting them out as in this PR, then i'd be happy to implement it that way instead. I could use `jobLabel` within the ServiceMonitor to pull that from a label's value in the service itself - but semantically i'm not sure what label key to apply to the service - something like `thanos-sidecar: thanos-sidecar` would seem a little odd for example.

Reply:
ok yeah makes sense, thanks for explaining! :)